### ISSUE TITLE
Modal cannot be opened when using Ionic prod build

### ISSUE THREAD

**Issue by Ritzlgrmft** [Ritzlgrmft/ionic-configuration-sample#1]
```text
Error: Uncaught (in promise): Error: No component factory found for t. Did you add it to @NgModule.entryComponents?
d@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/polyfills.js:3:3996
$@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:1:8765
resolveComponentFactory@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:2:29698
resolveComponentFactory@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:2:30111
_viewInit@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:16:19908
file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:16:17221
onInvoke@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:3:4435
run@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/polyfills.js:3:6474
file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/polyfills.js:3:3770
onInvokeTask@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:3:4339
runTask@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/polyfills.js:3:7163
a@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/polyfills.js:3:2319
invoke@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/polyfills.js:3:13254
dispatchEvent@[native code]
handleTapPolyfill@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:18:27339
pointerEnd@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:18:26060
[native code]
handleTouchEnd@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/58B88376-4E31-4459-9234-F428CF61FDC5/Configuration.app/www/build/main.js:20:11950
[native code] — polyfills.js:2
```

**Comment by Ritzlgrmft** [comment]
After disabling the minification, the error is slightly different:

```text
Error: Uncaught (in promise): Error: No component factory found for ModalCmp. Did you add it to @NgModule.entryComponents?
d@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/polyfills.js:3:3996
noComponentFactoryError@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:3580:39
resolveComponentFactory@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:3598:38
resolveComponentFactory@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:3640:110
_viewInit@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:49392:78
file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:49187:32
onInvoke@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:4403:43
run@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/polyfills.js:3:6474
file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/polyfills.js:3:3770
onInvokeTask@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:4394:47
runTask@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/polyfills.js:3:7163
a@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/polyfills.js:3:2319
invoke@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/polyfills.js:3:13254
dispatchEvent@[native code]
handleTapPolyfill@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:56342:36
pointerEnd@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:56228:35
[native code]
handleTouchEnd@file:///Users/markus/Library/Developer/CoreSimulator/Devices/C149A469-6EA1-435C-9DE4-05CF8E576A8E/data/Containers/Bundle/Application/66B491A3-5A42-4703-B108-9119101729E6/Configuration.app/www/build/main.js:66301:41
[native code] — polyfills.js:2
```

**Comment by Ritzlgrmft** [comment]
Problem is caused by tree-shaking. After adding
`"ionic_manual_treeshaking": false` everything works fine. Unfortunately, this is only a workaround.

**Comment by Ritzlgrmft** [comment]
Question posted to Ionic forum: https://forum.ionicframework.com/t/cannot-open-modal-from-another-module-in-prod-build/88458

**Comment by Ritzlgrmft** [comment]
Problem was caused by a bug in `@ionic/app-scripts@1.3.6`, which was fixed in `1.3.7` (see https://github.com/driftyco/ionic-app-scripts/issues/936)
